File: clojure/atbash-cipher/src/atbash_cipher.clj
(ns atbash-cipher (:require [clojure.string :as str]))

(def cipher
  (let [alphabet "abcdefghijklmnopqrstuvwxyz"
        numbers "0123456789"]
    (zipmap (concat alphabet numbers) (concat (reverse alphabet) numbers))))

(defn encode
  "Encodes text using the Atbash cipher."
  [plaintext]
  (->> 
    (str/lower-case plaintext)
    (map cipher)
    (filter some?)
    (partition-all 5)
    (map str/join)
    (str/join " ")))

(defn decode
  "Decodes text using the Atbash cipher."
  [ciphertext]
  (str/join (map cipher ciphertext)))

File: python/atbash-cipher/atbash_cipher.py
"""Implement the Atbash Cipher functions to encode and decode"""

ALPHABET = 'abcdefghijklmnopqrstuvwxyz'
NUMBERS = '0123456789'

CIPHER = dict(zip(ALPHABET + NUMBERS, ALPHABET[::-1] + NUMBERS))

def encode(plain_text):
    encoded = ''.join(CIPHER[char.lower()] for char in plain_text if char.isalnum())
    return ' '.join(encoded[index:index + 5] for index in range(0, len(encoded), 5))


def decode(ciphered_text):
    return ''.join(CIPHER[char] for char in ciphered_text if char.isalnum())
